### PR TITLE
Set referral code for embedded app

### PR DESCRIPTION
**Pull request by r-czajkowski**
Depends on: #760
Closes: #759 

This PR sets the referral code based on the embedded app.

### CODE REVIEW

File: dapp/src/hooks/useReferral.ts [hunk 1]
```diff
@@ -14,39 +12,19 @@ type UseReferralReturn = {
...
-    if (referralProgram.isValidReferral(convertedReferral)) {
-      setReferral(convertedReferral)
-    } else {
-      console.error("Incorrect referral")
-      setReferral(null)
-      openModal(MODAL_TYPES.UNEXPECTED_ERROR, {
-        withCloseButton: false,
-        closeOnEsc: false,
-      })
-    }
```
Comment (kkosiorowska) [inline dapp/src/hooks/useReferral.ts:48]: If you remove this part, the referral for the standalone dApp is not set correctly. Are we sure we want to disable the referral program for the standalone page?

File: dapp/src/components/Header/ConnectWallet.tsx
Comment (kkosiorowska) [inline dapp/src/components/Header/ConnectWallet.tsx:1]: It seems that changes for this file should be in https://github.com/thesis/acre/pull/760

**Comment by nkuba** [comment]
Does this PR have to use [`add-change-acc-btn`](https://github.com/thesis/acre/tree/add-change-acc-btn) as the base branch? Can't it go directly to the [`ledger-live-updates`](https://github.com/thesis/acre/tree/ledger-live-updates) branch?

File: dapp/src/components/Header/ConnectWallet.tsx
Comment (r-czajkowski) [reply dapp/src/components/Header/ConnectWallet.tsx:1]: I use `referralProgram.isEmbedApp(embeddedApp)` in `ConnectWallet` which I added in this PR. I can move the conditional rendering of the change account button to #760 and update the condition in this PR by using `referralProgram.isEmbedApp(embeddedApp)`.

**Comment by r-czajkowski** [comment]
> Does this PR have to use [`add-change-acc-btn`](https://github.com/thesis/acre/tree/add-change-acc-btn) as the base branch? Can't it go directly to the [`ledger-live-updates`](https://github.com/thesis/acre/tree/ledger-live-updates) branch?

Here we hide the change account button for the embedded app and use some of the functions I added in this PR.

File: dapp/src/components/Header/ConnectWallet.tsx
Comment (kkosiorowska) [reply dapp/src/components/Header/ConnectWallet.tsx:1]: Ah, okay. Let's leave it as it is.

File: dapp/src/components/Header/ConnectWallet.tsx [hunk 2]
```diff
@@ -111,22 +115,24 @@ export default function ConnectWallet() {
...
-            <IconButton
-              variant="ghost"
-              aria-label="Change account"
-              icon={<Icon as={IconUserCode} boxSize={5} />}
-              px={2}
-              boxSize={5}
-              onClick={() => handleConnectWallet(true)}
-            />
-          </Tooltip>
+          {isChangeAccountFeatureSupported(embeddedApp) && (
```
Comment (kkosiorowska) [inline dapp/src/components/Header/ConnectWallet.tsx:118]: Non-blocking comment

I think we don't need to create a separate function. We can simply use `isEmbed`.

**Review by kkosiorowska — APPROVED** [review]
I tested the solution and it seems to work well 🚀

File: dapp/src/components/Header/ConnectWallet.tsx (hunk 2 above)
Comment (r-czajkowski) [reply dapp/src/components/Header/ConnectWallet.tsx:118]: I think it depends on the embedded app. For example, we can support it in Ledger Live but not in Xverse.

**Pull request merged**